### PR TITLE
Explicitly pass scenario_id to http.send_file()

### PR DESCRIPTION
**Pull request by WillGibson**
I have been using this method to send log files to Report Portal and they re mostly, but not always being attached the the scenario that ran just before the one they belong to.

It is hoped that this change will allow us to specify the ID of the scenario they belong to.

I think I also managed to improve the tests for the `http.send_file()` method while I was working on it.

### CODE REVIEW

File: .rspec [hunk 1]
```diff
@@ -1 +1,2 @@
 --require spec_helper
+--format documentation
```
Comment (WillGibson) [inline .rspec:2]: Happy to revert this, but I found the more verbose output helpful.

File: lib/parallel_report_portal/http.rb [hunk 2]
```diff
@@ -2,24 +2,24 @@
 require 'tempfile'
 
 module ParallelReportPortal
-  # A collection of methods for communicating with the ReportPortal
```
Comment (WillGibson) [inline lib/parallel_report_portal/http.rb:5]: Oh, my IDE helped

File: lib/parallel_report_portal/http.rb [hunk 3]
```diff
@@ -202,19 +202,26 @@ def req_log(test_case_id, detail, level, time)
...
 
     # Request that Report Portal attach a file to the test case.
-    # 
+    #
     # @param status [String] the status level of the log, e.g. info, warn
     # @param path [String] the fully qualified path of the file to attach
     # @param label [String] a label to add to the attachment, defaults to the filename
     # @param time [Integer] the time in milliseconds for the attachment
     # @param mime_type [String] the mimetype of the attachment
-    def send_file(status, path, label = nil, time = ParallelReportPortal.clock, mime_type = 'image/png')
```
Comment (WillGibson) [inline lib/parallel_report_portal/http.rb:211]: This is where the actual changes start.

File: spec/parallel_report_portal/http_spec.rb [hunk 4]
```diff
@@ -89,28 +95,28 @@
     it 'issues a launch finished request' do
       stub_request(:put, "#{rp_endpoint}/launch/#{launch_id}/finish")
         .to_return(body: {id: launch_id}.to_json )
-
```
Comment (WillGibson) [inline spec/parallel_report_portal/http_spec.rb:92]: IDE helped again.

**Review by BillyRuffian — APPROVED** [review]

**Pull request merged**